### DISCUSSION TITLE
I would like a link to a 16bit DOS Example project (crosscompile linux to realmode DOS 8086)

### DISCUSSION THREAD

**Discussion by ccharon** [open-watcom/open-watcom-v2#1362] [Q&A]
Hi,

I would really appreciate some hints on a simple working project setup that has a Makefile which allows to build the project. Using Google and reading different pdfs, even asking ChatGPT I was not able to get an idea of how this could work in an orderly fashion.

On some pages even Cmake is mentioned which would be fine too. But searching did not give me any project that i could have a look at.

What am I looking for?
A simple project utilizing openwatcom to build a simple application under linux targeted at msdos 16bit realmode 8086. 

I would even create a project that resembles what I want and publish it on github if I only could find some ressources. that are more than install openwatcom, set environmet, call compiler. :P  this part I did manage, but it is inconvenient for a project.

**Comment by rdos314** [comment]
If you use the IDE, it's much easier. Just select "New project" under "File", and then select target "DOS 16-bit"

**Reply by jmalak** [reply]
If you create IDE msdos 16-bit realmode project (by example with IDE for Windows) then you can convert it to make file to build (cross-compile) on Linux or compile it on Windows in IDE.
OW has ide2make utility which read IDE project files and convert it to make files usable for selected build system.
Below is command line help

Usage: ide2make [*options] [tgtfile]*

  ide2make loads an IDE project file (-p switch) and, using the associated
  .tgt files, generates make files that can be invoked with WMAKE.
  If the .wpj file does not exist, a default project is used.  If any
  tgtfile(s) are specified, they are used as the targets in creating the
  make files.

options:
  -p wpjfile       - loads wpjfile.wpj (project.wpj by default).
  -c cfgfile       - loads cfgfile instead of ide.cfg.
  -d               - generate makefiles using development switch set.
  -r               - generate makefiles using release switch set.
  -h host type     - generate makefiles for selected host.
                     (default is current host)
                     0 - Windows 3.x
                     1 - OS/2 PM
                     2 - Windows NT
                     3 - Win-OS/2
                     4 - Windows 95
                     5 - Japanese Windows 3.x on an IBM
                     6 - Japanese Windows 3.x on a Nec98
                     7 - Dec Alpha (Windows NT)
                     8 - DOS
                     9 - Linux

In your case use your IDE file and convert it for Linux make files that you can compile it fully on Linux.
use **`-h 9`**

**Reply by ccharon** [reply]
I tried that and created a project with ide under windows. There I tried to create a makefile.

first I opened a terminal loaded, the environment variables with owsetenv.bat, then i switched to the directory of the project and executed:

ide2make -d -h 9 -p hello.wpj 

this worked without any message at all, but did not create a Makefile. 

I think I am missing something obvious here. Also wmake /? states in its help message that it would look for MAKEFILE.

**Reply by jmalak** [reply]
it should create **`project.mk`** and **`<target name>.mk1`** make files, one per each target in project.

you must run it by command
**`wmake -f project.mk`**

or you can rename **`project.mk`** make file to **`makefile`**
then you can run
**`wmake`**
